receiver/dcgmreceiver/scraper.go

```diff
@@ -41,13 +42,33 @@ func newDcgmScraper(config *Config, settings receiver.CreateSettings) *dcgmScrap
 	return &dcgmScraper{config: config, settings: settings}
 }
 
-func (s *dcgmScraper) start(_ context.Context, _ component.Host) error {
-	var err error
-	s.client, err = newClient(s.config, s.settings.Logger)
+// initClient will try to create a new dcgmClient if currently has no client;
+// it will try to initialize the communication with the DCGM service; if
+// success, create a client; only return errors if DCGM service is available but
+// failed to create client.
+func (s *dcgmScraper) initClient() error {
+	if s.client != nil {
+		return nil
+	}
+	if !isDcgmInstalled() {
+		s.settings.Logger.Warn("cannot initialize a DCGM client; DCGM is not installed.")
+		return nil
+	}
```

Comment:
It seems like you could move this block into `newClient` (or even, optionally, `initializeDcgm`) and have it return `ErrDcgmInitialization`. The behavior of the scraper would be identical, but the code would become simpler (callers wouldn't need to know about the separate "installed" check).

Reply:
Good point! Moved this check to `initializeDcgm()` and clean up the error message a bit, and the logic feels much clearer! Thanks!